### ISSUE TITLE
no-unused-expressions does not work

### ISSUE THREAD

**Issue by pronebird** [gajus/eslint-plugin-flowtype#348]
Hi,

I am not able to disable `no-unused-expressions`, I still get an error from ESLint for type assertion `(var: empty);`

```js
function getResult(status: Status): Result {
    switch (status) {
      case 'disconnected':
        return 'error';
      case 'connecting':
      case 'connected':
        return 'success';
      default:
        // Expected an assignment or function call and instead saw an expression  
        // no-unused-expressions
        (status: empty);
        throw new Error();
    }
  }
```

My `.eslintrc`:

```json
{
  "root": true,
  "extends": [
    "eslint:recommended",
    "plugin:react/recommended",
    "plugin:promise/recommended",
    "plugin:flowtype/recommended",
    "prettier",
    "prettier/flowtype",
    "prettier/react",
    "prettier/standard"
  ],
  "parser": "babel-eslint",
  "parserOptions": {
    "ecmaFeatures": {
      "jsx": true,
      "modules": true
    }
  },
  "plugins": [
    "react", 
    "flowtype", 
    "promise"
  ],
  "rules": {
    "prefer-const": "warn",
    "no-console": "off",
    "no-loop-func": "warn",
    "no-param-reassign": "warn",
    "func-names": "off",
    "no-unused-expressions": "error",
    "no-unused-vars": [
      "error",
      {
        "args": "all",
        "argsIgnorePattern": "_.*",
        "varsIgnorePattern": "_.*"
      }
    ],
    "block-scoped-var": "error",
    "react/prop-types": "off",
    "flowtype/define-flow-type": "warn",
    "flowtype/no-unused-expressions": "off"
  },
  "settings": {
    "react": {
      "pragma": "React",
      "version": "16.2",
      "flowVersion": "0.78"
    }
  },
  "env": {
    "es6": true,
    "node": true,
    "browser": true
  }
}
```

My `test/.eslintrc`:

```json
{
  "rules": {
    "no-unused-expressions" : "off",
    "react/no-render-return-value": "off"
  },
  "env": {
    "mocha": true
  },
  "globals": {
    "expect": true,
    "spy": true
  }
}
```

Now if I add `"flowtype/no-unused-expressions": "warn"` in the root `.eslintrc`, I get a barrage of warnings from `test/` folder which has it's own `.eslintrc` that explicitly disables `no-unused-expressions`. 

Stuff like this emits warnings:

```js
// warning  Expected an assignment or function call and instead saw an expression 
expect(rule.match('/other', '/route1')).to.be.null;
```

I can't really make any sense of it. Do I have certain configuration issue here or eslint-plugin-flowtype uses some hack to suppress unused expressions which results in false positives?

Versions:
```
eslint@4.19.1
babel-eslint@8.2.3
eslint-plugin-flowtype@2.50.0
```

**Comment by barretron** [comment]
I had the same issue. As far as I can tell, `flowtype/no-unused-expressions` is meant as a replacement for, rather than a modifier of `no-unused-expressions`. You have to go in and explicitly disable `no-unused-expressions` and enable `flowtype/no-unused-expressions` in its place. Similarly, you need to disable both versions if you want to disable the rule.

So, in your case:

`.eslintrc` should have:
```json
  "rules": {
    "no-unused-expressions": "off",
    "flowtype/no-unused-expressions": "error"
  }
```

and `test/.eslintrc` should have:
```json
  "rules": {
    "no-unused-expressions": "off",
    "flowtype/no-unused-expressions": "off"
  }
```

**Comment by gajus** [comment]
Closing as stale.
